### PR TITLE
[AppConfig]Default to standard sku if not specified 

### PR DESCRIPTION
**Pull request by shenmuxiaosen**
**Description**  

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppConfig

**Comment by shenmuxiaosen** [comment]
@avanigupta Can you help to take a look? Thanks.

**Comment by shenmuxiaosen** [comment]
Test failed due to an unrelated testcase: 
ERROR: test_private_endpoint_connection_private_link_scope (azure.cli.**command_modules.network**.tests.latest.test_private_endpoint_commands.NetworkPrivateLinkPrivateLinkScopeScenarioTest)

**Comment by yungezz** [comment]
hi @bim-msft could you pls have a look at this issue and the failed test? thanks

**Comment by avanigupta** [comment]
>                                             sku=Sku(name=sku))

Is the SKU name case sensitive?

---
Refers to: src/azure-cli/azure/cli/command_modules/appconfig/custom.py:35 in 0ac4179. [](commit_id = 0ac41798b1fdf1b1990ea068949500d562b28a39, deletion_comment = False)

**Comment by avanigupta** [comment]
>         string = construct_connection_string(cmd, config_store_name)

Since we are constructing the connection string here, it will always be "Endpoint={};Id={};Secret={}". But users could provide a connection string which is valid, but in a different format.
So we need to change the logic in line 71 and line 83 where we do string comparison of user provided connection string and our constructed connection string.

---
Refers to: src/azure-cli/azure/cli/command_modules/appconfig/_utils.py:68 in 0ac4179. [](commit_id = 0ac41798b1fdf1b1990ea068949500d562b28a39, deletion_comment = False)

File: src/azure-cli/azure/cli/command_modules/appconfig/_utils.py [hunk 1]
```diff
@@ -95,9 +95,10 @@ def is_valid_connection_string(connection_string):
         segments = connection_string.split(';')
         if len(segments) != 3:
             return False
-        if segments[0][:9] != 'Endpoint=' or segments[1][:3] != 'Id=' or segments[2][:7] != 'Secret=':
-            return False
-        return True
+
+        segments.sort()
+        if segments[0].startswith('Endpoint=') and segments[1].startswith('Id=') and segments[2].startswith('Secret='):
```
Comment (avanigupta) [inline src/azure-cli/azure/cli/command_modules/appconfig/_utils.py:100]: >'Endpoint=' [](start = 34, length = 11)

This has always worked well in the past, but to be on the safer side - do you know if connection strings are case sensitive? I wasn't able to find any concrete documentation about connection string formats.

**Comment by shenmuxiaosen** [comment]
>                                             sku=Sku(name=sku))

No. It is insensitive. Just to keep consistent with other services and examples we provided. 

---
In reply to: [659869071](https://github.com/Azure/azure-cli/pull/14398#issuecomment-659869071) [](ancestors = 659869071)

---
Refers to: src/azure-cli/azure/cli/command_modules/appconfig/custom.py:35 in 0ac4179. [](commit_id = 0ac41798b1fdf1b1990ea068949500d562b28a39, deletion_comment = False)

**Comment by shenmuxiaosen** [comment]
>         string = construct_connection_string(cmd, config_store_name)

Good catch. This needs to be fixed.

---
In reply to: [659873159](https://github.com/Azure/azure-cli/pull/14398#issuecomment-659873159) [](ancestors = 659873159)

---
Refers to: src/azure-cli/azure/cli/command_modules/appconfig/_utils.py:68 in 0ac4179. [](commit_id = 0ac41798b1fdf1b1990ea068949500d562b28a39, deletion_comment = False)

File: src/azure-cli/azure/cli/command_modules/appconfig/_utils.py (hunk 1 above)
Comment (shenmuxiaosen) [reply src/azure-cli/azure/cli/command_modules/appconfig/_utils.py:100]: Yeah, there is no hard requirement for the format of connection string. In most case, clients should get it from either CLI or portal and use it directly. However, we saw a customer reported issue in GitHub Action complaining for the order of those three segments. In CLI, didn't hear any complains about the format either order or case sensitivity. So I guess we leave it as case sensitive until there is a real ask. 

---
In reply to: [456233992](https://github.com/Azure/azure-cli/pull/14398#discussion_r456233992) [](ancestors = 456233992)

Comment (avanigupta) [reply src/azure-cli/azure/cli/command_modules/appconfig/_utils.py:100]: Sure, makes sense!

---
In reply to: [456708582](https://github.com/Azure/azure-cli/pull/14398#discussion_r456708582) [](ancestors = 456708582,456233992)

**Comment by yungezz** [comment]
hi @shenmuxiaosen is this PR ready to merge? cc @bim-msft

**Comment by shenmuxiaosen** [comment]
> hi @shenmuxiaosen is this PR ready to merge? cc @bim-msft

Yes. Please go ahead to merge it. Thanks.

**Pull request merged**